src/bun.js/api/bun/socket.zig

```diff
@@ -1144,7 +1151,10 @@ fn NewSocket(comptime ssl: bool) type {
                 return .zero;
             }
 
-            return this.writeOrEnd(globalObject, args.ptr[0..args.len], false);
+            return switch (this.writeOrEnd(globalObject, args.ptr[0..args.len], false)) {
+                .fail => .zero,
```

Comment:
We can’t return .zero to JavaScript unless an exception was thrown

Reply:
an exception was indeed thrown inside `.writeOrEnd()` for all the `.fail` cases...? :thinking: